### PR TITLE
Render branch cleanup

### PR DESCRIPTION
**Pull request by BigRoy**
Here are some proposed cleanups to the current code.

Note: Please double check the math logic etc.

- Simplified some arguments to functions
- Added docstrings/comments here and there
- Cleaned up redundant functions
- Avoid usage of private `._ordered` but use the `.ordered()` classmethod on some of the pxr usdview arg classes. 
- And also a lot of PEP08 cleanup

Let me know what you think - please comment if you disagree with certain choices.

**Note that I purely looked at the code - not at _where things are placed_ and whether the functionality in the end is what we need.**

### CODE REVIEW

File: usd_qtpy/render_util/dialog.py [hunk 1]
```diff
@@ -1,55 +1,61 @@
...
     REGEX_HASH = r"(#{2,})"
-    
-    hash_match = re.match(REGEX_HASH,filename)
+    hash_match = re.match(REGEX_HASH, filename)
     if not hash_match:
         filename = _rectify_path_framenumberspec(filename)    
     
     # ensure camera aspect ratio
-
+    # QUESTION: Why 24? Why that magic number?
```
Comment (Sasbom) [inline usd_qtpy/render_util/dialog.py:57]: 24 mm is a standard full frame SLR camera sensor size. (for the smallest size, so the height of the sensor).
[Wiki for reference](https://en.wikipedia.org/wiki/Image_sensor_format#Common_image_sensor_formats)
These are also the defaults for the usd thumbnail generator camera!

**Review by Sasbom — COMMENTED** [review]
I'll implement these manually after completing the turntable from file function.
Thank you!

File: usd_qtpy/render_util/dialog.py [hunk 2]
```diff
@@ -1,55 +1,61 @@
...
-        filter = "PNG (*.png);;JPEG (*.jpg, *.jpeg)"
+        caption="Save frame",
+        filter="PNG (*.png);;JPEG (*.jpg, *.jpeg)"
     )
 
-    camera = playblast.camera_from_stageview(stage,stageview,"viewerCamera")
+    camera = playblast.camera_from_stageview(stage, stageview, "viewerCamera")
 
     # ensure there's at least two or more hashes, if not, put in 4.
+    # QUESTION: Why do we need to enforce at least two hashes?
```
Comment (Sasbom) [inline usd_qtpy/render_util/dialog.py:50]: There's no particular reason, other than that the FrameRecorder API needs some hashes to put a timecode in.

I made an arbitrary but lightly informed decision that, if a "regular" image sequence you'd want to playblast is not going to be longer than 100 frames out of the tool, you'd do fine with having 2 decimal spots.
However, for a simple snapshot, I can't quite make that decision.

**Comment by Sasbom** [comment]
As of:
https://github.com/Sasbom/usd-qtpy/commit/880e37645335335adabebe0ef0354746dbc31125
nearly every cleanup has been implemented, and bugtested.

There are some things I kept in there, such as specifying the precision of transformation operations,
and some ordering which, when redone, gave some issues.

Thank you!

**Comment by BigRoy** [comment]
> such as specifying the precision of transformation operations,

Aren't those actually the default values you're passing? Basically, redundant arguments?

File: usd_qtpy/render_util/dialog.py (hunk 1 above)
Comment (BigRoy) [reply usd_qtpy/render_util/dialog.py:57]: A comment to clarify that would do wonders I'd say :)

**Comment by Sasbom** [comment]
> Aren't those actually the default values you're passing? Basically, redundant arguments?

The default precision for an operation added without arguments is XfromOp.PrecisionFloat, I ran a little experiment.

```py
>>> from pxr import Usd
>>> stage = Usd.Stage.CreateInMemory()
>>> from pxr import UsdGeom
>>> prim = UsdGeom.Xform.Define(stage,"/xform")
>>> op = prim.AddRotateXOp()
>>> op.GetPrecision()
UsdGeom.XformOp.PrecisionFloat
```
---
Apparently, different operation types have different standard values.
I just pass in the precision I'd like to store explicitly, for clarity.

```py
>>> op = prim.AddTranslateOp()
>>> op.GetPrecision()
UsdGeom.XformOp.PrecisionDouble
```

**Comment by BigRoy** [comment]
Do we care about the double precision? Why do we care about forcing a higher precision in this instance?

**Comment by Sasbom** [comment]
> Do we care about the double precision? Why do we care about forcing a higher precision in this instance?

I think so,
Assume someone needs to render a turntable that turns very very slowly, to showcase a large scene,
I would assume it's in best interest to keep the precision of the rotation as high as possible.

The values themselves are interpolated between frames, but ultimately rely on the precision of the place where the computed value of the operator is stored, so having floating point errors occur wouldn't be great, as sometimes the rotation spacing could be off.

I feel like specifying the precision to be the highest on the operator that is essential to a turn table,
just the one that turns stuff, is not a bad deal.
It'd be different we'd be dealing with a whole transformation matrix with 16 values, that'd be too much,
but a single rotation only has to store 1.

**Comment by BigRoy** [comment]
Fine with me, felt to me more like anticipating a problem that in reality wouldn't really become a quantifiable problem. I mean, what type of floating point precision's we are talking here, these are like 1e-6 values in degrees?

Anyway, don't care too much. Was just wondering whether we had good reason to explicitly set it higher, as if you were working around a current issue but that's not the case.

Thanks!

According to IBM:
> A single-precision float only has about 7 decimal digits of precision (actually the log base 10 of 223, or about 6.92 digits of precision)

I mean, rotating a 360 degrees with that would take about 3.6 billion frames to near the issue with precisions. Haha. I must have miscalculated that because that sounds insane. :')

**Comment by Sasbom** [comment]
> Fine with me, felt to me more like anticipating a problem that in reality wouldn't really become a quantifiable problem. I mean, what type of floating point precision's we are talking here, these are like 1e-6 values in degrees?
> 
> Anyway, don't care too much. Was just wondering whether we had good reason to explicitly set it higher, as if you were working around a current issue but that's not the case.
> 
> Thanks!

I just have nightmares about running into floating point precision errors while adjusting sliders in Houdini x)
That's about it, I don't wanna risk anything when it doesn't impact performance

**Pull request closed without merging**